src/main/java/frc/robot/util/SingleJointedMechanismVisualizer.java

```diff
@@ -0,0 +1,63 @@
+package frc.robot.util;
+
+import edu.wpi.first.math.geometry.Pose3d;
+import edu.wpi.first.math.geometry.Rotation2d;
+import edu.wpi.first.math.geometry.Rotation3d;
+import edu.wpi.first.wpilibj.smartdashboard.Mechanism2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismLigament2d;
+import edu.wpi.first.wpilibj.smartdashboard.MechanismRoot2d;
+import edu.wpi.first.wpilibj.util.Color;
+import edu.wpi.first.wpilibj.util.Color8Bit;
+import org.littletonrobotics.junction.Logger;
+
+public class SingleJointedMechanismVisualizer implements AutoCloseable {
+  private final String m_mechanismName;
+  private final String m_logKey;
+
+  private final Mechanism2d m_mechanism;
+  private final MechanismLigament2d m_mechanismLigament;
+
+  private final Pose3d m_pivotPose;
+
+  public SingleJointedMechanismVisualizer(
+      String mechanismName,
+      String logKey,
+      double ligamentLengthMeters,
+      Pose3d pivotPose,
+      Rotation2d defaultAngle) {
+    m_mechanismName = mechanismName;
+    m_logKey = logKey;
+
+    m_pivotPose = pivotPose;
+
+    m_mechanism = new Mechanism2d(2, 2, new Color8Bit(14, 17, 23));
```

Comment:
might make sense to make these parameters configurable because they rely on the size of the mechanism itself